### PR TITLE
feat(CDAP-20317): Add update time in repo config table and return in the GET /repository api

### PR DESCRIPTION
**Pull request by de-lan**
### What

1. Added the `updatedTimeMillis ` field in the repository table
2. Return the `updatedTimeMillis ` info in the GET api

### CODE REVIEW

File: cdap-proto/src/main/java/io/cdap/cdap/proto/sourcecontrol/RepositoryMeta.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  public RepositoryMeta(RepositoryConfig config, Long updateTime) {
+    this.config = new RepositoryConfig.Builder(config).build();
+    this.updateTime = updateTime;
+  }
+
+  public RepositoryConfig getConfig() {
+    return config;
+  }
+
+  public Long getUpdateTime() {
```
Comment (sanjanasandeep) [inline cdap-proto/src/main/java/io/cdap/cdap/proto/sourcecontrol/RepositoryMeta.java]: why is this Long object if it cannot be null?

Comment (de-lan) [reply cdap-proto/src/main/java/io/cdap/cdap/proto/sourcecontrol/RepositoryMeta.java]: Changed to type long

File: cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/SourceControlManagementHttpHandler.java [hunk 2]
```diff
@@ -61,23 +61,23 @@ public void setRepository(FullHttpRequest request, HttpResponder responder,
...
     try {
-      repoRequest.getRepository().validate();
+      repoRequest.getConfig().validate();
     } catch (InvalidRepositoryConfigException e) {
       responder.sendJson(HttpResponseStatus.BAD_REQUEST, GSON.toJson(new SetRepositoryResponse(e)));
       return;
     }
 
-    sourceControlService.setRepository(namespace, repoRequest.getRepository());
+    if (repoRequest.shouldTest()) {
```
Comment (de-lan) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/SourceControlManagementHttpHandler.java:76]: Moved the test connection code below the format validation block

File: cdap-proto/src/main/java/io/cdap/cdap/proto/sourcecontrol/RepositoryConfigRequest.java [hunk 3]
```diff
@@ -21,23 +21,23 @@
  */
 public class RepositoryConfigRequest {
   private final boolean test;
-  private final RepositoryConfig repository;
+  private final RepositoryConfig config;
```
Comment (de-lan) [inline cdap-proto/src/main/java/io/cdap/cdap/proto/sourcecontrol/RepositoryConfigRequest.java:24]: updated the name to be more consistent with table field

**Review by sanjanasandeep — APPROVED** [review]

File: cdap-proto/src/main/java/io/cdap/cdap/proto/sourcecontrol/RepositoryMeta.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+ */
+
+package io.cdap.cdap.proto.sourcecontrol;
+
+/**
+ * Metadata for repository configuration.
+ */
+public class RepositoryMeta {
+  private final RepositoryConfig config;
+  private final long updateTime;
```
Comment (albertshau) [inline cdap-proto/src/main/java/io/cdap/cdap/proto/sourcecontrol/RepositoryMeta.java]: nit: we use 'updatedTimeMillis' in the Drafts API, lets call it the same to be more consistent.

**Review by albertshau — APPROVED** [review]
minor comment, otherwise lgtm

File: cdap-proto/src/main/java/io/cdap/cdap/proto/sourcecontrol/RepositoryMeta.java (hunk 4 above)
Comment (de-lan) [reply cdap-proto/src/main/java/io/cdap/cdap/proto/sourcecontrol/RepositoryMeta.java]: updated to updatedTimeMillis

**Pull request merged**